sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs

```diff
@@ -36,12 +37,19 @@ public ClientTestBase(bool isAsync)
         [TearDown]
         public virtual void GlobalTimeoutTearDown()
         {
+            if (Debugger.IsAttached)
+            {
+                return;
+            }
+
             var executionContext = TestExecutionContext.CurrentContext;
             var duration = DateTime.UtcNow - TestStartTime;
-            if (duration > TimeSpan.FromSeconds(GLOBAL_TEST_TIMEOUT_IN_SECONDS) && !Debugger.IsAttached)
+            var timeout = TestEnvironment.GlobalIsRunningInCI ? GLOBAL_TEST_TIMEOUT_IN_SECONDS : GLOBAL_LOCAL_TEST_TIMEOUT_IN_SECONDS;
```

Comment:
What about when you are running the CI command locally?

Reply:
I'm not sure what you mean by running the CI command locally. The way the GlobalIsRunningInCI works it to look for an env variable that should only be present on a CI host.

Reply:
```
dotnet test eng/service.proj --filter "(TestCategory!=Manually) & (TestCategory!=Live)" --framework netcoreapp3.1 --logger "trx;LogFileName=netcoreapp3.1.trx" --logger:"console;verbosity=normal" /p:SDKType=all /p:ServiceDirectory=* /p:IncludeSrc=false /p:IncludeSamples=false /p:IncludePerf=false /p:IncludeStress=false /p:RunApiCompat=false /p:InheritDocEnabled=false /p:Configuration=Debug /p:CollectCoverage=false /p:CodeCoverageDirectory=/mnt/vss/_work/1/s\sdk\core /p:UseProjectReferenceToAzureClients=true
```

If I run this locally, I'd want the higher timeout, but I will get the local timeout.

Reply:
> If I run this locally, I'd want the higher timeout, but I will get the local timeout.

Can you expand more on why you'd want longer timeouts? As long as it is still running on your local hardware, it should still be much faster per test than the CI host.

Reply:
Do we have evidence that this is the case? I've seen similar test durations in my local runs.
In any case, people are likely to rerun tests locally until they pass rather than changing the test. I don't see how this change is reducing flakiness.

Reply:
While it may be true in general that the tests will be faster locally, any tests that may take over 5s locally but are reliably under 10s would stop being reliable after this change. There would be no way to detect such tests from running the CI in this PR.

Reply:
> Do we have evidence that this is the case? I've seen similar test durations in my local runs. In any case, people are likely to rerun tests locally until they pass rather than changing the test. I don't see how this change is reducing flakiness.

Mostly just through observation. The CI hosts are generally much more thread-bound than any of our local dev machines. 

But the sort of flakiness that local test failures would create is exactly why we would want this change in the first place. A local failure is much quicker to deal with than waiting for it to fail in CI (and it is non-blocking).

Generally speaking, no non-Live test should take more than 5 seconds locally. Do we have any examples of tests that would fail locally with this limit?

Reply:
> While it may be true in general that the tests will be faster locally, any tests that may take over 5s locally but are reliably under 10s would stop being reliable after this change. There would be no way to detect such tests from running the CI in this PR.

That's ok - because locally failing tests due to timeout won't block anyone from merging a PR. And if they do, they need to be fixed anyway because they are likely to fail in CI intermittently in the future.